Remove problematic characters from feature dirs

xylar opened the pull request:
This merge removes characters that are sometimes problematic from the directory names of features.

This merge also fixes some incorrect docstrings, adding a missing component and correcting the use of tags for reading features.

pep8speaks commented:
Hello @xylar! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`geometric_features/geometric_features.py`](https://github.com/MPAS-Dev/geometric_features/blob/c7ab9eee5d325c716d79a50a8bbaf7790e99b2fa/geometric_features/geometric_features.py):

> [Line 83:81](https://github.com/MPAS-Dev/geometric_features/blob/c7ab9eee5d325c716d79a50a8bbaf7790e99b2fa/geometric_features/geometric_features.py#L83): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 80 characters)
> [Line 176:81](https://github.com/MPAS-Dev/geometric_features/blob/c7ab9eee5d325c716d79a50a8bbaf7790e99b2fa/geometric_features/geometric_features.py#L176): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 80 characters)
> [Line 226:81](https://github.com/MPAS-Dev/geometric_features/blob/c7ab9eee5d325c716d79a50a8bbaf7790e99b2fa/geometric_features/geometric_features.py#L226): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (100 > 80 characters)
> [Line 301:81](https://github.com/MPAS-Dev/geometric_features/blob/c7ab9eee5d325c716d79a50a8bbaf7790e99b2fa/geometric_features/geometric_features.py#L301): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 80 characters)

xylar commented:
@proteanplanet, this should make it safe to use the feature names you originally had, in case you want to do that.

xylar commented:
## Testing

I merged all the features and split them again, proving that none of the existing names is affected by removing these potential problem characters.  Here was my script:

```python
#!/usr/bin/env python
from geometric_features import GeometricFeatures, FeatureCollection


gf = GeometricFeatures(cacheLocation='./geometric_data')

fc = FeatureCollection()
for component in ['bedmachine', 'bedmap2', 'iceshelves', 'landice',
                  'natural_earth', 'ocean']:
    for objectType in ['point', 'transect', 'region']:
        if objectType in gf.allFeaturesAndTags[component]:
            print(component, objectType)
            fc.merge(gf.read(componentName=component, objectType=objectType))


gf = GeometricFeatures(cacheLocation='./new_geometric_data')
gf.split(fc)
```
